### ISSUE TITLE
Q: standardise tooltips?

### ISSUE THREAD

**Issue by samschlicht** [talentbeyondboundaries/tbbtalentv2#1753]
Strictly speaking (for accessibility reasons), all tooltips should be clickable — but even if we stick with hover-over, I think they should all be the same style.

The new style makes use of NgbTooltip functionality, which is a good thing to make standard. Furthermore, I find that the old style take so long to appear that you could easily not know they exist.

Personally I think this one's a no-brainer and should probably be addressed before employer access beta testing, which I put to the group>

Old style:

<img width="1080" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeEluQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--bd8335c4a2e6500727ec9a0c157dddaf3eda0f13/Screenshot%202024-12-16%20at%2011.47.35%E2%80%AFam.png" alt="Screenshot 2024-12-16 at 11.47.35 am.png" />

Increasingly prevalent new style:

<img width="500" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeE1uQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--0a57221862e250bf7911d517318c14ee52cb99b3/Screenshot%202024-12-16%20at%2011.47.48%E2%80%AFam.png" alt="Screenshot 2024-12-16 at 11.47.48 am.png" />

**Comment by sadatmalik** [comment]
I think there's a case for both tooltips and hover-overs, but I agree that applying consistent styling to both makes sense.

Looking at the AWS console as an example, they use clickable tooltip-style "info" buttons next to headings (opening a sidebar in their case) and hover-overs for more graphical elements like cost charts. 

If we go for clickable tooltips in all cases, do we risk cluttering the UI with too many little "?" icons? 

But yes, consistent styling is ideal, it might be more of a nice-to-have rather than a prerequisite for employer access.

**Comment by samschlicht** [comment]
@sadatmalik agreed that it needn't be a prerequisite for Employer Access. In fact, I think this is probably most properly included in the Accessibility and User Experience epics.

Probably the right balance re hover-over vs clickable depends on necessity: if the tooltip provides information that is essential to using the platform, then making it non-clickable excludes keyboard-only users. If it's a value-add, then hover-over is probably acceptable.

Happy to leave that discussion for those epics - in the meantime, whether any new tooltips required for Employer Access are clickable or hover-over can be approached on a case-by-case basis. I'll certainly make them ngbTooltips as the old style ones are barely detectable.

**Comment by cazcam34** [comment]
@sadatmalik  I have wondered this too @samschlicht, but I agree I think the ngbTooltip could be a bit too 'in your face' especially when we have small icons next to each other.  I think the case by case basis is a good one.  However the regular hover over is pretty slow, it would be nice for it to work a bit quicker.
In terms of making the style consistent could overwrite the ngbTooltip to be lightish grey with black text like the standard hover over - or vice versa. Perhaps this can go in the next release though?

**Comment by samschlicht** [comment]
Yea that's a fair point, @cazcam34 — if there's some kind of guidance as to when we use one or the other, then having a lighter-touch option makes sense. It definitely needs to appear quicker, though — I'd say at the moment that it takes long enough to appear that most users aren't getting the benefit.

Absolutely agree that this would be a candidate for the next release: it may sit best with the implementation of some of our redesigned screens, since the styling would ideally reflect those.

**Comment by cazcam34** [comment]
@hiba-machfej when reviewing the User Experience epic I found this in there that I think could go into our redesign UI epic. Could be worth thinking about how we use and style popovers in the redesign?

**Comment by cazcam34** [comment]
Moving to phase 2
